storage/pebble/registers.go

```diff
@@ -61,7 +69,14 @@ func (s *Registers) Store(
 	height uint64,
 	entries flow.RegisterEntries,
 ) error {
-	//  TODO: add value check once LatestHeight is implemented
+	// value check
+	firstHeight, latestHeight, err := s.getHeightRange()
+	if err != nil {
+		return fmt.Errorf("unable to perform height validation: %w", err)
+	}
+	if height > latestHeight+1 || height < firstHeight {
```

Comment:
We should have a check that makes sure the height is either the same as the last height or increased by one, we shouldn't be overwriting older registers, don't see the use case for it. 
So something like:
```
if height - latestHeight > 1 { error } 
```